### DISCUSSION TITLE
How to add dependencies like firebase or others ?

### DISCUSSION THREAD

**Discussion by DominatorVbN** [skiptools/skip#10] [Q&A] [answered]
How we can go about adding commonly used open sourced decencies, i am assuming there would be an bridging required for iOS variant of lib and android version of lib, very similar to SwiftUI and Compose.

Can we get more direction toward this ?

**Comment by marcprux** [comment, marked as answer]
Good question. I'll first state outright that adding an external SPM package as a dependency to your Skip project will only work if that dependency itself is Skip-enabled. 

So, for example, just adding AlamoFire or Firebase dependencies won't work, because they won't know which libraries to talk to on the Android side. 

We are building an ecosystem of Skip-enabled frameworks that will satisfy most common needs for modern apps (see https://skip.tools/docs/modules/), but it is still in a very early state. However, you can make your own Skip-enabled frameworks, just like we do: see the [Framework Development](https://skip.tools/docs/gettingstarted/#framework_development) section of the Gettin Started guide. 

When interfacing with external libraries on the Android side, you will add the correct Android/Java dependencies to your `skip.yml` file (see https://github.com/skiptools/skip-script/blob/main/Sources/SkipScript/Skip/skip.yml for an example), and then your Swift code will use `#if SKIP` blocks to delegate based on the platform (e.g., see https://github.com/skiptools/skip-foundation/blob/main/Sources/SkipFoundation/Data.swift for some very simple examples of how we bridge `Foundation.Data` to `kotlin.ByteArray`). 

This is an advanced topic that is currently under-documented, but we're excited to help out and resolve any issues you encounter.

**Comment by Invader1999** [comment]
Same Question what does it mean by create own framework how to even do it can we  reate the skip-firebase if yes what is the process

**Reply by Invader1999** [reply]
It would be good if we had a sample step by step documentation or video on how to create own skip - framework . If people like me get an Idea how it's done many frameworks can be added
